lib/packwerk/application_validator.rb

```diff
@@ -34,19 +31,10 @@ def check_all
       merge_results(results)
     end
 
-    def check_autoload_path_cache
-      expected = Set.new(@application_load_paths)
-      actual = Set.new(@configuration.load_paths)
-      if expected == actual
-        Result.new(true)
-      else
-        Result.new(
-          false,
-          "Load path cache in #{@config_file_path} incorrect!\n"\
-          "Paths missing from file:\n#{format_yaml_strings(expected - actual)}\n"\
-          "Extraneous load paths in file:\n#{format_yaml_strings(actual - expected)}"
-        )
-      end
+    def validate_rails_app!
```

Comment:
I decided to have the API to this here even though the implementation is in configuration (and the error is raised in the application load paths class), since it is a validator after all. Happy to incorporate any feedback here.

Reply:
I think we're talking about three responsibilities here:

1. making sure we have a Rails app
2. booting the Rails app
3. extracting load paths

`ApplicationLoadPaths` currently does all of that. It might make sense to move `1` and `2` into a separate class, to be reused here? Or have a differently named class that does it all - but I think the load path extraction logic is actually interesting enough to deserve its own class.

Reply:
I think we may not need to boot the application for `validate` anymore once we have removed the two application state caches, so it might actually make sense to _not_ extract the load paths here.

Reply:
Looking at this again, I think this comment still applies.

Do we actually need to validate it's a rails app here? If we don't `packwerk check` might fail, but it would fail with a pretty obvious error message.